src/main/java/io/vertx/core/http/impl/HttpClientRequestImpl.java

```diff
@@ -440,59 +440,64 @@ protected String hostHeader() {
   }
 
   private synchronized void connect(Handler<HttpVersion> headersHandler) {
-    if (!connecting) {
+    if (connecting) {
```

Comment:
can you avoid changing the control flow in the PR, so the differences with the actual code are easy to see, otherwise it really adds lot of noise to the actual review work

Reply:
@vietj done. Sorry about that. I just thought that would make the code more readable. Fixed in the last commit

Reply:
these should be done in separate PR's